dev_guide/dev_tutorials/binary_builds.adoc

```diff
@@ -0,0 +1,307 @@
+  oc start-build ruby-hello-world --from-dir="." --follow
+  ----
+
+Once your build has completed and the application has redeployed, navigating to the application's route host should 
+result in a page with a blue background.
+
+You can keep making changes locally and building your code with `oc start-build --from-dir`.
+
+You can also create a branch of the code, commit your changes locally, and use the repository's HEAD as the source
+for your build:
+ 
+----
+git checkout -b my_branch
+git add .
+git commit -m "My changes"
+oc start-build ruby-hello-world --from-repo="." --follow
+----
+
+
+[[binary-builds-private-code]]
+=== Tutorial: Building private code
+
+. Create a local directory to hold your code:
++
+----
+mkdir myapp
+cd myapp
+----
+
+. In the directory create a file named `Dockerfile` with the following content:
++
+----
+FROM centos:centos7
+
+EXPOSE 8080
+
+COPY index.html /var/run/web/index.html
+
+CMD cd /var/run/web && python -m SimpleHTTPServer 8080
+----
+
+. Create a file named `index.html` with the following content:
++
+----
+<html>
+  <head>
+    <title>My local app</title>
+  </head>
+  <body>
+    <h1>Hello World</h1>
+    <p>This is my local application</p>
+  </body>
+</html>
+----
+
+. Create a new build for your application:
++
+----
+oc new-build --strategy docker --binary --docker-image centos:centos7 --name myapp
+----
+
+. Start a binary build using the local directory's content:
++
+----
+oc start-build myapp --from-dir . --follow
+----
+
+. Deploy the application using new-app, then create a route for it:
++
+----
+oc new-app myapp
+oc expose svc/myapp
+----
+
+. Get the host name for your route and navigate to it:
++
+----
+oc get route myapp
+----
+
+After having built and deployed your code, you can iterate by making changes to your local
+files and starting new builds by invoking `oc start-build myapp --from-dir .`. Once built,
+the code will be automatically deployed and the changes will be reflected in your browser when you refresh the page.
+
+[[binary-builds-pipeline-binary-artifacts]]
+=== Tutorial: Binary artifacts from pipeline
+
+Jenkins on OpenShift allows using slave images with the appropriate tools to build your code. For example,
+you can use the `maven` slave to build a WAR from your code repository. However, once this artifact is built,
+you need to commit it to an image that contains the right runtime artifacts to run your code. A binary build
+may be used to add these artifacts to your runtime image. In the following tutorial, we'll create a Jenkins
+pipeline that makes use of the maven slave to build a WAR, and then uses a binary build with a Dockerfile to
+add that WAR to a wildfly runtime image.
+
+. Create a new directory for your application:
++
+----
+mkdir mavenapp
+cd mavenapp
+----
+
+. Create a Dockerfile that copies a WAR to the appropriate location inside a wildfly image for execution. Copy the following
+  to a local file named `Dockerfile`:
++
+----
+FROM wildfly:latest
```

Comment:
openshift/wildfly-110-centos7?

not sure what wildfly:latest is.

Reply:
I added a note with an explanation of why I'm using it. I think it's a good opportunity to educate about `new-app/new-build` magic.